### PR TITLE
[clang-tidy] initial checkin

### PR DESCRIPTION
**Pull request by msbaines**
Just checking one directory for now. Will send follow-up PRs to
fix other directories.

Issue: #323

## Motivation and Context

Automatically enforce coding style.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [  ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by mosra** [comment]
Forgive me if the answer is "yes of course" but since I never used Clang Tidy, I have to ask -- does this check function/variable/... naming as well? Such as the case here: https://github.com/facebookresearch/habitat-sim/pull/309/files#r340844325

Thanks! :)

**Comment by msbaines** [comment]
> Forgive me if the answer is "yes of course" but since I never used Clang Tidy, I have to ask -- does this check function/variable/... naming as well? Such as the case here: https://github.com/facebookresearch/habitat-sim/pull/309/files#r340844325
> 
> Thanks! :)

Not yet but should be possible to add. Looks like google coding style is still WiP upstream. The list of available checks so far:

https://clang.llvm.org/extra/clang-tidy/checks/list.html

File: run-clang-tidy.py [hunk 1]
```diff
@@ -0,0 +1,388 @@
+#!/usr/bin/env python
```
Comment (erikwijmans) [inline run-clang-tidy.py]: Can we move this to a `tools` or something folder?

**Comment by msbaines** [comment]
All feedback addressed. Ready for review.

File: tools/run-clang-tidy.py [hunk 2]
```diff
@@ -0,0 +1,388 @@
+#!/usr/bin/env python
```
Comment (mathfac) [inline tools/run-clang-tidy.py]: @msbaines is this custom script for Habitat or it can be `pip` installed?

Comment (msbaines) [reply tools/run-clang-tidy.py]: The script comes from here:

https://github.com/llvm-mirror/clang-tools-extra/blob/master/clang-tidy/tool/run-clang-tidy.py

I don't believe its available via pip install.

On Mac, you will also need to:

brew install llvm

**Pull request merged**